### PR TITLE
Add fluid added mass to inertial

### PR DESCRIPTION
**Pull request by chapulina**
# 🎉 New feature

* Part of https://github.com/gazebosim/gz-sim/issues/1462
* Depends on
    * https://github.com/gazebosim/sdformat/pull/1076
    * https://github.com/gazebosim/gz-math/pull/459

## Summary

Implements the [Fluid Added Mass Proposal](http://sdformat.org/tutorials?tut=added_mass_proposal&branch=chapulina/added_mass).

## Test it

See the added tests. I'll be opening downstream PRs next.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: sdf/1.10/inertial.sdf [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <element name="inertial" required="0">
   <description>
     The link's mass, position of its center of mass, and its central inertia
-    properties.
+    properties. Optionally, it's fluid added mass.
```
[the remark this answers is not in the document]
Comment (chapulina) [reply sdf/1.10/inertial.sdf]: a5de3cb1a87c97f720f322077d0f6893e2bed212

File: sdf/1.10/inertial.sdf [hunk 2]
```diff
@@ -88,4 +88,120 @@
...
   </element> <!-- End Inertia -->
+
+  <element name="fluid_added_mass" required="0">
+    <description>
+      This link's fluid added mass matrix about the link's center of mass.
+      This matrix represents the inertia of the fluid that is dislocated when the
+      body moves. Added mass should be zero if the density of the surrounding
+      fluid is negligible with respect to the body's density.
+      The 6x6 matrix is symmetric, therefore only 21 unique elements can be set.
+      The elements of the matrix follow the [x, y, z, p, q, r] notation.
```
[the remark this answers is not in the document]
Comment (chapulina) [reply sdf/1.10/inertial.sdf]: a5de3cb1a87c97f720f322077d0f6893e2bed212

File: sdf/1.10/inertial.sdf [hunk 3]
```diff
@@ -88,4 +88,120 @@
...
+      </description>
+    </element>
+    <element name="zr" type="double" default="0.0" required="0">
+      <description>
+        Added mass in the Z axis due to angular acceleration about the Z axis, and vice-versa.
+      </description>
+    </element>
+    <element name="pp" type="double" default="0.0" required="0">
+      <description>
+        Added mass moment about the X axis due to angular acceleration about the X axis, and vice-versa.
```
[the remark this answers is not in the document]
Comment (chapulina) [reply sdf/1.10/inertial.sdf]: a5de3cb1a87c97f720f322077d0f6893e2bed212

File: sdf/1.10/inertial.sdf [hunk 4]
```diff
@@ -88,4 +88,120 @@
...
+      </description>
+    </element>
+    <element name="qr" type="double" default="0.0" required="0">
+      <description>
+        Added mass moment about the Y axis due to angular acceleration about the Z axis, and vice-versa.
+      </description>
+    </element>
+    <element name="rr" type="double" default="0.0" required="0">
+      <description>
+        Added mass moment about the Z axis due to angular acceleration about the Z axis, and vice-versa.
```
[the remark this answers is not in the document]
Comment (chapulina) [reply sdf/1.10/inertial.sdf]: a5de3cb1a87c97f720f322077d0f6893e2bed212

File: src/Link.cc [hunk 5]
```diff
@@ -66,7 +66,7 @@ class sdf::Link::Implementation
   /// \brief The inertial information for this link.
   public: gz::math::Inertiald inertial {{1.0,
             gz::math::Vector3d::One, gz::math::Vector3d::Zero},
-            gz::math::Pose3d::Zero};
+            gz::math::Pose3d::Zero, gz::math::Matrix6d::Zero};
```
[the remark this answers is not in the document]
Comment (chapulina) [reply src/Link.cc]: a5de3cb1a87c97f720f322077d0f6893e2bed212

File: src/Link.cc [hunk 6]
```diff
@@ -182,6 +230,13 @@ Errors Link::Load(ElementPtr _sdf)
                      this->Name() +
                      " has invalid inertia."});
   }
+  if (!this->dataPtr->inertial.SetFluidAddedMass(addedMass))
```
[the remark this answers is not in the document]
Comment (chapulina) [reply src/Link.cc]: a5de3cb1a87c97f720f322077d0f6893e2bed212

File: sdf/1.10/inertial.sdf [hunk 7]
```diff
@@ -2,7 +2,7 @@
 <element name="inertial" required="0">
   <description>
     The link's mass, position of its center of mass, and its central inertia
```
Comment (scpeters) [inline sdf/1.10/inertial.sdf:4]: nit: remove `and` from "and its central inertia"

**Review by quarkytale — APPROVED** [review]
Looks good to me!

**Pull request merged**